pipeline-runner/R/gem2s-2-load_user_files.R

```diff
@@ -71,19 +72,38 @@ read_10x_files <- function(config, input_dir) {
       paste(sample_fpaths, collapse = " - ")
     )
 
-    counts <- Seurat::Read10X(sample_dir, gene.column = 1)
+    annot <- read.delim(annot_fpath, header = FALSE)
 
-    if (is(counts, "list")) {
-      slot <- "Gene Expression"
-      # questionable: grab first slot if no gene expression
-      if (!slot %in% names(counts)) slot <- names(counts)[1]
-      counts <- counts[[slot]]
+    if (ncol(annot) == 1 || annot[1, 2] == "Gene Expression") {
```

Comment:
All of this code dealing with the annotations is very hard to read to me or any engineer without the actual seurat object. First I'd suggest to move all inside the loop intro 3 different functions: one for producing the `annotations` and the other to produce the `counts_list`. `e.g.`:
```
    annot <- extract_annotations(annot_fpath)
    features_types <- extract_feature_types(annot)
    counts <- extract_counts(sample_dir, annot, feature_types)
```

Reply:
Keep in mind that at this stage there's no seurat object, we are just working with a table with 2-3 columns

Reply:
I originally wanted to refactor this for into functions, and I almost pushed that change but eventually decided that it's simpler to read in this way if you are following the flow of what's going on with the tables. @gerbeldo what do you think?

We can certainly do this if both of you agree that it will look better.

Reply:
I am refactoring a bit, given the conflicts with #233.

Reply:
Also adding some comments to the obscure parts, such as this.